src/toGamut.js

```diff
@@ -179,21 +179,24 @@ export function toGamutCSS (origin, { space = origin.space }) {
 	}
 	let min = 0;
 	let max = origin_OKLCH.coords[1];
-
 	let min_inGamut = true;
-	let current;
+	let current = clone(origin_OKLCH);
+    let clipped = clip(current);
+
+    let E = deltaEOK(clipped, current);
+    if (E < JND) {
+		return clipped;
+	}
```

Comment:
@jamesnw I think this is what was missing here (and also missing from the spec?). Before entering the `while` loop, both the codepen and ColorAide are _first_ checking if the delta between the color and the clipped version is less than the `JND`, and if so returning the clipped version.

@svgeesus Should this be added to the spec?

Reply:
Yes I think this should be added

Reply:
Now added: https://github.com/w3c/csswg-drafts/commit/06748226bb73a7fee401ba0d00c4e4655ec17555